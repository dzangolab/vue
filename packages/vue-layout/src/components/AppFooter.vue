<template>
  <footer>
    <slot name="copyright">
      <Copyright
        v-if="config?.copyright"
        :holder="config.copyright.holder"
        :url="config.copyright.url"
      />
    </slot>
    <slot name="social"> </slot>
    <slot name="version">
      <div class="version">
        {{ config.appVersion }}
      </div>
    </slot>
  </footer>
</template>

<script lang="ts">
export default {
  name: "AppFooter",
};
</script>

<script setup lang="ts">
import { useConfig } from "@dzangolab/vue3-config";

import Copyright from "./Copyright.vue";

import type { AppConfig } from "@dzangolab/vue3-config";

const config = useConfig() as AppConfig;
</script>
