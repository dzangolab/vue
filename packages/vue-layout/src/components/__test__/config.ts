import type { AppConfig } from "@dzangolab/vue3-config";
import type { LocaleMessages, VueMessageType } from "@dzangolab/vue3-i18n";

const config: AppConfig = {
  apiBaseUrl: "http://localhost",
  appTitle: "Test app",
  appVersion: "0.1.0",
  copyright: {
    holder: "Dzango Technologies Limited",
    url: "https://dzango.com",
  },
  i18n: {
    fallbackLocale: "en",
    messages: {} as LocaleMessages<VueMessageType>,
    supportedLocales: ["en", "fr"],
  },
  features: {
    showVersion: true,
    sentry: true,
  },
  layout: {
    mainMenu: [
      {
        name: "Home",
        route: "home",
      },
      {
        name: "About",
        route: "about",
      },
    ],
  },
  sentry: {
    dsn: "tbd",
  },
  slug: "app",
  websiteDomain: "test",
};

export default config;
