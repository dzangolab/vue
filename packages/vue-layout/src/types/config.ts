import type { MenuItem } from "./menu";
import type { AppConfig } from "@dzangolab/vue3-config";
import type { LocaleMessages, VueMessageType } from "@dzangolab/vue3-i18n";

interface DzangolabVue3LayoutConfig {
  mainMenu?: MenuItem[];
}

interface DzangolabVue3LayoutPluginOptions {
  config: AppConfig;
  translations?: LocaleMessages<VueMessageType>;
}

export type { DzangolabVue3LayoutConfig, DzangolabVue3LayoutPluginOptions };
