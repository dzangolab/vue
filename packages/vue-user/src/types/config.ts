import { DefineComponent } from "vue";

import type { RouteOverrides } from "./router";
import type {
  IsEmailOptions,
  StrongPasswordOptions,
} from "@dzangolab/vue3-form";

interface DzangolabVueUserConfig {
  features?: {
    signUp?: {
      emailVerification?: boolean;
      termsAndConditions?: {
        display?: boolean;
        label: DefineComponent;
        showCheckbox?: boolean;
      };
    };
    updateEmail?: boolean;
  };
  password?: {
    minLength: number;
  };
  options?: {
    email?: IsEmailOptions;
    password?: StrongPasswordOptions;
  };
  routes?: RouteOverrides;
  socialLogins?: string[];
  supportedRoles?: string[];
}

declare module "@dzangolab/vue3-config" {
  export interface AppConfig {
    user?: DzangolabVueUserConfig;
  }
}

export type { DzangolabVueUserConfig };
