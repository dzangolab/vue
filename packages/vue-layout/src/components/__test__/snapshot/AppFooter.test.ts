import configPlugin from "@dzangolab/vue3-config";
import { shallowMount } from "@vue/test-utils";
import { describe, expect, it } from "vitest";

import AppFooter from "@/components/AppFooter.vue";

import config from "../config";

import type { VueWrapper } from "@vue/test-utils";

describe("AppFooter", () => {
  it("matches snapshot", () => {
    const wrapper: VueWrapper = shallowMount(AppFooter, {
      global: {
        plugins: [
          [
            configPlugin,
            {
              config,
            },
          ],
        ],
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
