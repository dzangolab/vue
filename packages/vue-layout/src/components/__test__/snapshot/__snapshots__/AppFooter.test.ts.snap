// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AppFooter > matches snapshot 1`] = `
<footer>
  
  <copyright-stub
    holder="Dzango Technologies Limited"
    showdisclaimer="false"
    url="https://dzango.com"
  />
  
  
  
  
  <div
    class="version"
  >
    0.1.0
  </div>
  
</footer>
`;
